src/daos_vol.c

```diff
@@ -89,7 +89,13 @@ static herr_t H5_daos_fill_def_plist_cache(void);
 static void * H5_daos_faccess_info_copy(const void *_old_fa);
 static herr_t H5_daos_faccess_info_free(void *_fa);
 static herr_t H5_daos_get_conn_cls(void *item, H5VL_get_conn_lvl_t lvl, const H5VL_class_t **conn_cls);
-static herr_t H5_daos_get_cap_flags(const void *info, unsigned *cap_flags);
+
+#if H5VL_VERSION > 2
```

Comment:
Nitpick: I think it'd be clearer to say H5VL_VERSION >= 3. It seems more intuitive to say something was changed in v3 than in "the version after v2"